code/game/data_huds.dm

```diff
@@ -314,3 +317,87 @@
 			holder.icon_state = "hudmove"
 		else
 			holder.icon_state = ""
+
+/*~~~~~~~~~~~~~~
+	PLANT HUD
+~~~~~~~~~~~~~~~*/
+/proc/RoundPlantBar(value)
+	switch(value * 100)
+		if(91 to INFINITY)
+			return "max"
+		if(81 to 90)
+			return "90"
+		if(71 to 80)
+			return "80"
+		if(61 to 70)
+			return "70"
+		if(51 to 60)
+			return "60"
+		if(41 to 50)
+			return "50"
+		if(31 to 40)
+			return "40"
+		if(21 to 30)
+			return "30"
+		if(11 to 20)
+			return "20"
+		if(1 to 10)
+			return "10"
+		else
+			return "zero"
+	return "zero"
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/plant_hud_set_nutrient()
+	var/image/holder = hud_list[PLANT_NUTRIENT_HUD]
+	holder.icon_state = "hudnutrient[RoundPlantBar(nutrilevel/maxnutri)]"
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/plant_hud_set_water()
+	var/image/holder = hud_list[PLANT_WATER_HUD]
+	holder.icon_state = "hudwater[RoundPlantBar(waterlevel/maxwater)]"
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/plant_hud_set_status()
+	var/image/holder = hud_list[PLANT_STATUS_HUD]
+	// Dis be drugs here. And special mention plants
+	var/list/special = list(/datum/seed/nettle, /datum/seed/apple/poison, /datum/seed/ambrosia, /datum/seed/tobacco, /datum/seed/mushroom, \
+							/datum/seed/weeds, /datum/seed/grass, /datum/seed/kudzu, /datum/seed/diona, /datum/seed/clown)
+	if (!seed)
+		holder.icon_state = ""
+		return
+	if (harvest)
+		if (is_type_in_list(seed, special))
+			holder.icon_state = "hudharvest2"
```

Comment:
Also interesting. When the plant is mutated b bees the seed type just gets turned into a generic datum/seed instead of what the plant was. I guess I can't use this then. Darn